src/components/program-progress/data/utils.jsx

```diff
@@ -31,3 +32,67 @@ export function getProgramCertImage(type) {
       return '#';
   }
 }
+
+export function getLastEndingCourseDate(courses) {
```

Comment:
Isn't this utility supposed to return end date of a course run falling at the end, unsure why we are passing courses here instead of a single course.

Reply:
All notCompleted courses are passed to this function. It will check all courses and course runs and return the lastEndingCourseDate